src/pymor/models/iosys.py

```diff
@@ -147,6 +165,26 @@ def __init__(self, A, B, C, D=None, E=None, sampling_time=0, presets=None,
         sampling_time = float(sampling_time)
         assert sampling_time >= 0
 
+        assert T is None or T > 0
+
+        if T is not None:
+            if initial_data is None:
+                initial_data = A.source.zeros(1)
+            if isinstance(initial_data, VectorArray):
+                assert initial_data in A.source
+                assert len(initial_data) == 1
+                initial_data = VectorOperator(initial_data, name='initial_data')
+            assert initial_data.source.is_scalar
+            assert initial_data.range == A.source
+
+            if sampling_time == 0:
+                assert isinstance(time_stepper, TimeStepper)
+                assert not isinstance(time_stepper, DiscreteTimeStepper)
+            else:
+                if time_stepper is None:
+                    time_stepper = DiscreteTimeStepper()
+                assert isinstance(time_stepper, DiscreteTimeStepper)
```

Comment:
What should happen when a `time_stepper` is provided but `T` is `None`?

Reply:
As I mentioned above, there is no time-stepping if `T` is `None`. Maybe an exception could be raised if `time_stepper` is given by `T` is not? (It might be nice if `T` was a part of `time_stepper`.)

Reply:
Yes, I would raise an exception in that case. - Not sure if `T` should be only part of the `time_stepper`. If we write down a intial-value problem we usually also specify the time-interval for which we seek a solution. So from my point of view this is part of the `Model`.

Reply:
An exception is now raised.

> Not sure if `T` should be only part of the `time_stepper`. If we write down a intial-value problem we usually also specify the time-interval for which we seek a solution. So from my point of view this is part of the `Model`.

For LTI systems, I don't remember seeing `T` specified except for time-limited MOR methods. Usually, `T` is taken to be $\infty$.